routes/routes.js

```diff
@@ -0,0 +1,215 @@
+import express from 'express';
+import fs from 'fs';
+import { randomUUID } from 'crypto';
+import { users, carts, orders } from '../storage.js';
+import { EventEmitter } from 'events';
+import multer from 'multer';
+import csv from 'csv-parser';
+import Joi from 'joi';
+import products from '../storage/products.store.json' assert { type: 'json' };
+
+const router = express.Router();
+const upload = multer({ dest: 'uploads/' });
+const eventEmitter = new EventEmitter();
+
+eventEmitter.on('fileUploadStart', () => {
+  logEvent('File upload has started');
+});
+
+eventEmitter.on('fileUploadEnd', () => {
+  logEvent('File has been uploaded');
+});
+
+eventEmitter.on('fileUploadFailed', (error) => {
+  logEvent(`Error occurred, file upload failed; Details: ${error.message}`);
+});
+
+const logEvent = (message) => {
+  const date = new Date();
+  const formattedDate = date.toLocaleString('en-GB', {
+    year: 'numeric',
+    month: '2-digit',
+    day: '2-digit',
+    hour: '2-digit',
+    minute: '2-digit',
+    second: '2-digit',
+    hour12: false
+  }).replace(',', '');
+
+  const logMessage = `${formattedDate} - ${message}\n`;
+  fs.appendFileSync('storage/filesUpload.log', logMessage);
+};
+
+const productSchema = Joi.object({
+  name: Joi.string().required(),
+  description: Joi.string().required(),
+  category: Joi.string().required(),
+  price: Joi.number().required(),
+}).unknown(true);
+
+const userSchema = Joi.object({
+  email: Joi.string().email().required().messages({
+    'string.email': 'Invalid email format',
+    'any.required': 'Email is required',
+  }),
+  name: Joi.string().required().messages({
+    'any.required': 'Name is required',
+  }),
+  password: Joi.string()
+    .min(8)
+    .pattern(new RegExp('^(?=.*[A-Z])(?=.*[a-z])(?=.*\\d)(?=.*[?!@#$%^&*])[A-Za-z\\d?!@#$%^&*]{8,}$'))
+    .required()
+    .messages({
+      'string.min': 'Password must be at least 8 characters long',
+      'string.pattern.base': 'Password must include upper and lower case letters, numbers, and special characters',
+      'any.required': 'Password is required',
+    }),
+});
+
+
+
+router.post('/register', (req, res) => {
+  try {
+    const { email, name, password } = req.body;
+    const { error } = userSchema.validate({ email, name, password });
+
+    if (error) {
+      return res.status(400).json({ error: error.details[0].message });
```

Comment:
On the lecture we were considering examples of creating custom classes for error, implement implementation from the lecture or similar